### PR TITLE
feat: improve error behaviour && styles

### LINKED ISSUE #107: Custom amount visual QA
- the error messages shouldn't cause the button to move ideally - space allocated for it, smaller size. unfortunately main every.org seems to have regressed here but will post mocks of our standard input shortly.
- enter button doesn't work for custom amount.
- when you enter a valid amount, the error message should disappear.

- screencast https://app.usebubbles.com/9jDzPb6QpF9CTd5phaxfKj/comments-on-donate-button-pvjdwsg2n-vercel-app
- design system (let me know if you need access, if so which email account) https://www.figma.com/file/vukSu4ISAHKCAqX6UOzXZ6/Every-Design-System?node-id=123%3A2418
- screenshot of design system inputs

Screen Shot 2021-02-15 at 16 17 19

as of now it seems there's space for this. We can also make this input style reusable.

_Originally posted by @osdiab in https://github.com/everydotorg/donate-button/issues/100#issuecomment-779002348_

### PR DESCRIPTION
**Pull request by martinbianchi**
closes #107 

@osdiab The only thing missing is to style the input equal to the one that your already have. 
I do not have access to your design sistem on figma, If you think that will look better using the colors from there, please grant me access to this email martin@p22.studio and I will change it

### CODE REVIEW

**Comment by osdiab** [comment]
Looking good! A couple updates:

- Border looks chopped off on my phone
- because font size in the input is too small it seems to cause the screen to zoom in on iOS when you tap it for accessibility reasons. Would be best if input font size wasn't smaller than default.

Will send you access so you can get exact pixel sizes, colors etc 

image

**Comment by osdiab** [comment]
added access!

**Comment by martinbianchi** [comment]
@osdiab I updated the PR and fixed those issues. Also I adjusted the input error state styles to match with your designs.

File: packages/donate-button/src/components/Donations/DonateButton/index.tsx [hunk 1]
```diff
@@ -66,6 +66,20 @@ const DonateButton = ({
 	const formText = monthlyDonation ? lang.monthly : lang.oneTime;
 	const donationAmount = donationsContextValue?.donationAmount;
 
+	useEffect(() => {
```
Comment (osdiab) [inline packages/donate-button/src/components/Donations/DonateButton/index.tsx:69]: you can actually do this by making the actual donate form be a form, and then the button have `type="submit"`; then the browser will kick in and do all the accessibility magic to make the expected stuff just work out of the box. will make a separate ticket for that.

**Review by osdiab — APPROVED** [review]

**Comment by osdiab** [comment]
the layout still moves a bit when the error message shows; but can fix that up later, not a big deal.

**Pull request merged**